### PR TITLE
FEATURE: closed events

### PR DESCRIPTION
**Pull request by ZogStriP**
Adds support for "closed" events instead of just relying on the start & end dates.

An event will be "expired" when it happened in the past.

An event might now also be "closed", if for whatever reasons, the author decides to "cancel" it.

Context - https://meta.discourse.org/t/when-closing-event-it-moves-it-to-todays-date-time/307292

### CODE REVIEW

File: app/serializers/discourse_post_event/event_serializer.rb [hunk 1]
```diff
@@ -2,33 +2,34 @@
 
 module DiscoursePostEvent
   class EventSerializer < ApplicationSerializer
-    attributes :id
```
Comment (ZogStriP) [inline app/serializers/discourse_post_event/event_serializer.rb:5]: I just sorted alphabetically all the attributes.

Only added "`is_closed`"

File: assets/javascripts/discourse/widgets/discourse-post-event-dates.js [hunk 2]
```diff
@@ -15,27 +15,28 @@ export default createWidget("discourse-post-event-dates", {
...
     const content = [
       iconNode("clock"),
-      h("span.date", new RawHtml({ html: `<span>${attrs.localDates}</span>` })),
+      h("span.date", new RawHtml({ html: `<span>${localDates}</span>` })),
     ];
 
     if (
-      attrs.eventModel.is_expired &&
-      attrs.eventModel.status !== "standalone"
+      eventModel.is_expired &&
```
Comment (ZogStriP) [inline assets/javascripts/discourse/widgets/discourse-post-event-dates.js:25]: Ensures we only show participants when the event was not "standalone" nor closed.

File: spec/system/post_event_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    composer.submit
+
+    expect(page).to have_content(title)
+
+    page.find("#more-dropdown").click
+    page.find(".item-closeEvent").click
+    page.find("#dialog-holder .btn-primary").click
+
+    expect(page).to have_content("Closed")
```
Comment (pmusaraj) [inline spec/system/post_event_spec.rb:33]: Nitpick, do we need this to be more specific? i.e. page should have an element with a given classname and content "Closed".

File: spec/system/post_event_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    page.find("#more-dropdown").click
+    page.find(".item-openEvent").click
+    page.find("#dialog-holder .btn-primary").click
+
+    expect(page).to have_content("Public")
```
Comment (pmusaraj) [inline spec/system/post_event_spec.rb:39]: Probably useful to ensure that page no longer has "Closed" here.

**Review by pmusaraj — APPROVED** [review]

**Pull request merged**